File: frontend/src/data/skills.ts
export interface Skill {
  name: string
  level: number // 0-100
  category: 'frontend' | 'backend' | 'tools'
}

export const skills: Skill[] = [
  // Frontend
  { name: 'JavaScript', level: 95, category: 'frontend' },
  { name: 'Vue.js', level: 90, category: 'frontend' },
  { name: 'React', level: 85, category: 'frontend' },
  { name: 'HTML', level: 95, category: 'frontend' },
  { name: 'CSS/SASS', level: 90, category: 'frontend' },
  { name: 'Redux', level: 75, category: 'frontend' },

  // Backend
  { name: 'Node.js', level: 80, category: 'backend' },
  { name: 'Express', level: 75, category: 'backend' },
  { name: 'REST APIs', level: 85, category: 'backend' },
  { name: 'Socket.io', level: 70, category: 'backend' },
  { name: 'Firebase', level: 75, category: 'backend' },

  // Tools
  { name: 'Git', level: 85, category: 'tools' },
  { name: 'Vite', level: 80, category: 'tools' },
  { name: 'TypeScript', level: 85, category: 'tools' }
]

export const skillsByCategory = {
  frontend: skills.filter(s => s.category === 'frontend'),
  backend: skills.filter(s => s.category === 'backend'),
  tools: skills.filter(s => s.category === 'tools')
}

File: backend/knowledge/skills.py
from knowledge.models import Skill

skills_data = {
    "frontend": [
        Skill(name="JavaScript", level=95, category="frontend"),
        Skill(name="Vue.js", level=90, category="frontend"),
        Skill(name="Vue 3 / Composition API", level=90, category="frontend"),
        Skill(name="React", level=85, category="frontend"),
        Skill(name="TypeScript", level=85, category="frontend"),
        Skill(name="HTML5", level=95, category="frontend"),
        Skill(name="CSS3 / SASS", level=90, category="frontend"),
        Skill(name="Nuxt.js", level=75, category="frontend"),
        Skill(name="Redux", level=75, category="frontend"),
        Skill(name="Responsive Web Design", level=90, category="frontend"),
    ],
    "backend": [
        Skill(name="Node.js", level=80, category="backend"),
        Skill(name="Express.js", level=75, category="backend"),
        Skill(name="REST APIs", level=85, category="backend"),
        Skill(name="Socket.io", level=70, category="backend"),
        Skill(name="Firebase", level=75, category="backend"),
        Skill(name="MongoDB", level=75, category="backend"),
        Skill(name="MySQL", level=70, category="backend"),
    ],
    "tools": [
        Skill(name="Git / GitHub", level=90, category="tools"),
        Skill(name="Vite", level=85, category="tools"),
        Skill(name="Webpack", level=80, category="tools"),
        Skill(name="Pinia", level=85, category="tools"),
        Skill(name="PrimeVue", level=80, category="tools"),
        Skill(name="Docker", level=70, category="tools"),
        Skill(name="Figma", level=75, category="tools"),
    ]
}

# Flat list of all skills for easy searching
all_skills = skills_data["frontend"] + skills_data["backend"] + skills_data["tools"]
